src/name_service/delivery/utils.clj

```diff
@@ -0,0 +1,47 @@
+(ns name-service.delivery.utils
+  (:require [cheshire.core :as json]
+            [io.pedestal.http :as bootstrap]
+            [io.pedestal.http.body-params :as body-params]
+            [name-service.core.action :as action]
+            [name-service.delivery.http :as http]))
+
+(defn map-key-first [map]
+  (-> map
+      keys
+      first
+      name))
+
+(defn map-value-first [map]
+  (-> map
+      vals
+      first))
+
+(defn merge-with-payload->json [coll payload]
+  (->>
+    coll
+    (merge payload)
+    (json/encode)))
+
+(defn inject-interceptors [service-map]
+  (-> service-map
+      bootstrap/default-interceptors
+      (update ::bootstrap/interceptors conj (body-params/body-params))
+      (update ::bootstrap/interceptors conj bootstrap/html-body)))
+
+(defn action->response
+  "Converts an action to a response."
+  [{:keys [::action/type ::action/error ::action/payload]}]
+  (let [type-name (name type)]
+    (cond
+      (= error :key-does-not-exist)
+      (http/bad-request (json/encode {:error "Key does not exist." :type type-name}))
+      (= error :key-exists)
+      (http/bad-request (json/encode {:error "Key exist." :type type-name}))
+      (= error :storage)
+      (http/bad-request (json/encode {:error  "Error occurred on storage action."
```

Comment:
This error message reveals too much about implementation.
Use something like "Saving data failed"